frontend/public/components/cluster-settings/request-header-idp-form.tsx

```diff
@@ -0,0 +1,218 @@
+    this.setState({clientCommonNames});
+  };
+
+  headersChanged = (headers: string[]) => {
+    this.setState({headers});
+  };
+
+  preferredUsernameHeadersChanged = (preferredUsernameHeaders: string[]) => {
+    this.setState({preferredUsernameHeaders});
+  };
+
+  nameHeadersChanged = (nameHeaders: string[]) => {
+    this.setState({nameHeaders});
+  };
+
+  emailHeadersChanged = (emailHeaders: string[]) => {
+    this.setState({emailHeaders});
+  };
+
+  caFileChanged = (caFileContent: string) => {
+    this.setState({caFileContent});
+  };
+
+  render() {
+    const { name, challengeURL, loginURL, caFileContent } = this.state;
+    const title = 'Add Identity Provider: Request Header';
+    return <div className="co-m-pane__body">
+      <Helmet>
+        <title>{title}</title>
+      </Helmet>
+      <form onSubmit={this.submit} name="form" className="co-m-pane__body-group co-m-pane__form">
+        <h1 className="co-m-pane__heading">{title}</h1>
+        <p className="co-m-pane__explanation">
+          Use request header to identify users from request header values. It is typically used in combination with an authenticating proxy, which sets the request header value.
+        </p>
+        <IDPNameInput value={name} onChange={this.nameChanged} />
+        <div className="co-form-section__separator" />
+        <h3>URLs</h3>
+        <p className="co-m-pane__explanation">
+          At least one URL must be provided.
+        </p>
+        <div className="form-group">
+          <label className="control-label" htmlFor="challenge-url">Challenge URL</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.challengeURLChanged}
+            value={challengeURL}
+            id="challenge-url"
+            aria-describedby="challenge-url-help" />
+          <div className="help-block" id="challenge-url-help">
+            The URL to redirect unauthenticated requests from OAuth clients which expect interactive logins.
+          </div>
+        </div>
+        <div className="form-group">
+          <label className="control-label" htmlFor="login-url">Login URL</label>
+          <input className="form-control"
+            type="text"
+            onChange={this.loginURLChanged}
+            value={loginURL}
+            id="login-url"
+            aria-describedby="login-url-help" />
+          <div className="help-block" id="login-url-help">
+            The URL to redirect unauthenticated requests from OAuth clients which expect WWW-Authenticate challenges.
+          </div>
+        </div>
+        <div className="co-form-section__separator" />
+        <h3>More Options</h3>
+        <IDPCAFileInput value={caFileContent} onChange={this.caFileChanged} isRequired />
+        <ListInput label="Client Common Names" onChange={this.clientCommonNamesChanged} />
+        <div className="help-block">
+          An optional list of common names to require a match from.
```

Comment:
This should be an optional prop to the `ListInput` component. Then we can set the right `aria-describedby` headers for accessibility. Also we need to fix up the spacing. I'd suggest putting the description either immediately below the input or between the label and input.

Reply:
@spadgett yeah i thought you said to add this in the follow on pr?

Reply:
If we do it as a follow on, I'd take the descriptions out of this PR.

Reply:
ok

Reply:
@spadgett removed help until follow on